### PR TITLE
FIO imx6ullevk changes

### PR DESCRIPTION
**Pull request by mike-scott**
This series splits out SPL and u-boot for the imx6ullevk platform.

Several drivers had to be adjusted so they could be disabled for SPL due to SPL size limitations

### CODE REVIEW

File: arch/arm/include/asm/arch-mx6/imx-regs.h [hunk 1]
```diff
@@ -317,6 +317,7 @@
 #define I2C4_BASE_ADDR              (AIPS2_OFF_BASE_ADDR + 0x78000)
 #define IP2APB_USBPHY1_BASE_ADDR    (AIPS2_OFF_BASE_ADDR + 0x78000)
 #define IP2APB_USBPHY2_BASE_ADDR    (AIPS2_OFF_BASE_ADDR + 0x7C000)
+
```
Comment (mike-scott) [inline arch/arm/include/asm/arch-mx6/imx-regs.h]: Guh.  Not sure how this snuck in.

**Comment by mike-scott** [comment]
@ricardosalveti 
Changes include:
- Typo fix in https://github.com/foundriesio/u-boot/pull/26/commits/c1eb520fadcac2cfb3ba452d0e848fb3eb1a03f8 (fasboot -> fastboot)
- removed `#if 0` blocks from https://github.com/foundriesio/u-boot/pull/26/commits/946bb660ca15d2e2d1ed660a18c59fa016b0a786
- dropped accidental newline in https://github.com/foundriesio/u-boot/pull/26/commits/18236d66d79305b683b5e72214d2e02279e31465
- dropped the patches which added / changed the following vars: script_addr, fdt_addr, optee_ovl_addr
- use default LOADADDR and dropped the patch allowing it to be set differently

File: arch/arm/mach-imx/syscounter.c [hunk 2]
```diff
@@ -55,7 +55,7 @@ static inline unsigned long long us_to_tick(unsigned long long usec)
 	return usec;
 }
 
-#ifndef CONFIG_SKIP_LOWLEVEL_INIT
+#if !defined(CONFIG_SKIP_LOWLEVEL_INIT) || defined(CONFIG_SPL_BUILD)
```
Comment (mike-scott) [inline arch/arm/mach-imx/syscounter.c:58]: Not entirely sure if this will have some sort of spill over into other IMX platforms building SPL

**Comment by mike-scott** [comment]
Testing this from scratch now on local builds due to memory changes

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**